src/Cache/Engine/RedisEngine.php

```diff
@@ -320,22 +356,45 @@ public function deleteAsync(string $key): bool
      */
     public function clear(): bool
     {
-        $this->_Redis->setOption(Redis::OPT_SCAN, (string)Redis::SCAN_RETRY);
-
         $isAllDeleted = true;
         $iterator = null;
         $pattern = $this->_config['prefix'] . '*';
+        $scanCount = (int)$this->_config['scanCount'];
+
+        // Set scan option only if using a Redis instance (not RedisCluster)
+        if ($this->_Redis instanceof Redis) {
+            $this->_Redis->setOption(Redis::OPT_SCAN, Redis::SCAN_RETRY);
+        }
 
-        while (true) {
-            $keys = $this->_Redis->scan($iterator, $pattern, (int)$this->_config['scanCount']);
+        if ($this->_Redis instanceof RedisCluster) {
+            foreach ($this->_Redis->_masters() as $masterNode) {
+                $iterator = null;
 
-            if ($keys === false) {
-                break;
-            }
+                while (true) {
+                    $keys = $this->_Redis->scan($iterator, $masterNode, $pattern, $scanCount);
+
+                    if ($keys === false) {
+                        break;
+                    }
 
-            foreach ($keys as $key) {
-                $isDeleted = ((int)$this->_Redis->del($key) > 0);
-                $isAllDeleted = $isAllDeleted && $isDeleted;
+                    foreach ($keys as $key) {
+                        $isDeleted = ($this->_Redis->del($key) > 0);
+                        $isAllDeleted = $isAllDeleted && $isDeleted;
+                    }
+                }
+            }
```

Comment:
Return `$isDeleted` here after the `foreach` loop and remove the `else` usage.

Reply:
We follow the "return early" principle (which reduces code branches). Adding a return statement to the `if ($this->_Redis instanceof RedisCluster) {`  block would avoid the need for using an `else` block for dealing with case of `Redis` instance.

Reply:
I think I got what you mean now. You want to only have one if block for the Redis cluster and return `$isAllDeleted` in that and putting the normal handling below it without the need for else.
Your suggestion also had that it should return `$isDeleted` but the current Redis code returns `$isAllDeleted` so that we will return true even when there were no cache keys to be found.